test/ts/compile-ts.tsx

```diff
@@ -232,14 +270,34 @@ class InjectSomeStores extends Component<{ x: any }, {}> {
 inject(({ x }) => ({ x }))(InjectSomeStores)
 
 // TODO: not possible: App2.wrappedComponent
+{
+    class T extends Component<{ x: number }> {
+        render() {
+            return <div />
+        }
+    }
+
+    const Injected = inject("test")(T)
+    ;<Injected.wrappedComponent x={3} />
+}
 
 {
-	class T extends Component<{x: number}> {
-		render() {
-			return <div/>
-		}
-	}
-
-	const Injected = inject("test")(T);
-	<Injected.wrappedComponent x={3} />
-}
\ No newline at end of file
+    // just to make sure it compiles
```

Comment:
It would be great if we had an actual Typescript test for this, as the decorator implementation of TS is different from the babel one, but it might be too much of a hassle to add it to the current setup, I am confident enough that this case is simple enough to not fail (famous last words)